src/open-feature.ts

```diff
@@ -0,0 +1,59 @@
+import { OpenFeatureClient } from './client';
+import { NOOP_PROVIDER } from './no-op-provider';
+import { Client, Context, EvaluationLifeCycle, FlagValue, Hook, Provider } from './types';
+
+// use a symbol as a key for the global singleton
+const GLOBAL_OPENFEATURE_API_KEY = Symbol.for('@openfeature/js.api');
+
+type OpenFeatureGlobal = {
+  [GLOBAL_OPENFEATURE_API_KEY]?: OpenFeature;
+};
+const _global = global as OpenFeatureGlobal;
```

Comment:
I could have done this in a separate file, but then I'd have to export it and it would be exported in the dist. Keeping it in this file makes it effectively private without introducing a bundler, which we may want to do later.